notification-server/src/bin/api.rs

```diff
@@ -31,20 +40,71 @@ struct Args {
     log_level:      log::LevelFilter,
 }
 
-#[derive(Clone)]
+#[derive(Debug)]
 struct AppState {
-    db_connection: Config,
+    db_connection: DatabaseConnection,
+}
+
+const MAX_RESOURCES_LENGTH: usize = 1000;
+
+lazy_static! {
+    static ref MAX_PREFERENCES_LENGTH: usize = all::<Preference>().collect::<Vec<_>>().len();
 }
 
+#[tracing::instrument]
```

Comment:
Also if you have large arguments the logs produced become very hard to read. I recommend filtering the information by skipping and focusing relevant information:

https://docs.rs/tracing/latest/tracing/attr.instrument.html#skipping-fields

Could for example do
```suggestion
#[tracing::instrument(skip(state))]
```